NitroxServer/GameLogic/EventTriggerer.cs

```diff
@@ -30,31 +45,56 @@ internal set
             }
         }
 
+        /// <summary>
+        /// Total Elapsed Time in seconds
+        /// </summary>
         public double ElapsedSeconds
         {
             get => ElapsedTimeMs * 0.001;
             private set => ElapsedTimeMs = value * 1000;
         }
 
+        /// <summary>
+        /// Subnautica's equivalent of days
+        /// </summary>
         // Using ceiling because days count start at 1 and not 0
         public int Day => (int)Math.Ceiling(ElapsedTimeMs / TimeSpan.FromMinutes(20).TotalMilliseconds);
 
-        public EventTriggerer(PlayerManager playerManager, string seed, double elapsedTime, double? auroraExplosionTime)
+        public EventTriggerer(PlayerManager playerManager, PDAStateData pdaStateData, StoryGoalData storyGoalData, string seed, double elapsedTime, double? auroraExplosionTime, double? auroraWarningTime)
         {
             this.playerManager = playerManager;
+            this.pdaStateData = pdaStateData;
+            this.storyGoalData = storyGoalData;
+            this.seed = seed;
             // Default time in Base SN is 480s
             elapsedTimeOutsideStopWatchMs = elapsedTime == 0 ? TimeSpan.FromSeconds(480).TotalMilliseconds : elapsedTime;
+            // The timer interval is in milliseconds, and so the AuroraExplosionTime should be
```

Comment:
```suggestion
```
If this is really important the variable should be called auroraExplosionTimeMs and not a "hidden" comment inside the constructor.